### ISSUE TITLE
add list of namespaces to itsyouonline.Client

### ISSUE THREAD

**Issue by iwanbk** [zero-os/0-stor#505]
So user knows all namespaces he has

**Comment by GlenDC** [comment]
I haven't checked the itsyou.online REST API, but I suppose if it supports it we can indeed easily support it as well, and add it to the Daemon's Namespace Service and as zerostor's namespace sub commands as well.

Can you please specify however why you need this feature. Our dev resources are limited and spread at the moment, so for the initial launch I would like to keep the amount of features we have to implement to a minimum. Therefore I would like to read a proper motivation, backed by evidence that this feature is absolutely required. Otherwise I'll move it to roadmap instead.

**Comment by iwanbk** [comment]
> Can you please specify however why you need this feature.

We need it for minio backend, similar to #502, it is very useful when user browsing his files using browser or minio client.

I would like it to be implemented if IYO supported it.

**Comment by GlenDC** [comment]
Seems like itsyou.online supports it:

```raml
/{username}/organizations:
    securedBy: [oauth_2_0: { scopes: [ "user:admin" ] } ]
    get:
      displayName: GetUserOrganizations
      description: Get the list organizations a user is owner or member of
      responses:
        200:
          body:
            application/json:
              properties:
                owner: string[]
                member: string[]
```
> full RAML spec: https://itsyou.online/apidocumentation/raml/itsyouonline.raml

We'll just have to do some trickery to figure out the actual namespaces, as that's _all_ organizations of a user, which could also be non 0-stor related. The easiest and cheapest way to do so, is probably to filter the name, as we use a very specific format for our (root) namespace organizations.

Either way, this one is fairly simple as far as I can see, so I'll leave this open for the TF Launch milestone.

**Comment by iwanbk** [comment]
I can see two IYO APIs to  achieve this.

The first one already mentioned on https://github.com/zero-os/0-stor/issues/505#issuecomment-362559279. There are some issues with this API:
- it includes organizations not under organization specified in 0-stor's `organization` config.
- we need to add `username` field in 0-stor IYO config

Other API is `/organizations/{globalid}/tree`
```
 /tree:
      securedBy: [oauth_2_0: { scopes: [ "organization:owner", "organization:member" ] } ]
      get:
        displayName: GetOrganizationTree
        description: Tree structure of all suborganizations
        responses:
          200:
            body:
              application/json:
                type: OrganizationTreeItem[]
```
It doesn't have issues with previous API, however it currently has bug, tracked at https://github.com/itsyouonline/identityserver/issues/706, i guess it should be quick to be fixed.
 
I prefer to use `GetOrganizationTree` API because of the issues in `GetUserOrganizations` API mentioned before

**Comment by iwanbk** [comment]
> it includes organizations not under organization specified in 0-stor's organization config.

We can easily filter it though.

your call @GlenDC

**Comment by GlenDC** [comment]
Yeah let's go for option 2, where we make use of the `GetOrganizationTree` method. Please track itsyouonline/identityserver#706, and let us know when that issue is fixed, so we can integrate this feature in 0-stor for you.

**Comment by iwanbk** [comment]
> Please track itsyouonline/identityserver#706, and let us know when that issue is fixed, so we can integrate this feature in 0-stor for you.

fixed

**Comment by iwanbk** [comment]
> so we can integrate this feature in 0-stor for you.

you can assign to me if you want me to work on it.

**Comment by GlenDC** [comment]
Closed in 6161103add81bfb3b2f6377c1e97a3b4445dd95a
